Resources/Prototypes/Entities/Objects/Consumable/Food/frozen.yml

```diff
@@ -0,0 +1,236 @@
+# Base
+
+- type: entity
+  parent: BaseItem
+  id: FoodFrozenBase
+  abstract: true
+  components:
+  - type: Food
+  - type: Sprite
+    sprite: Objects/Consumable/Food/frozen.rsi
+    netsync: false
+  - type: LoopingSound
+  - type: SolutionContainer
+    maxVol: 20 # For sprinkles or something? Idk.
+    contents:
+      reagents:
+      - ReagentId: Nutriment
+        Quantity: 10
+
+# Ice cream
+
+- type: entity
+  name: ice cream sandwich
+  parent: FoodFrozenBase
+  id: FoodFrozenSandwich
+  description: Portable Ice-cream in its own packaging.
+  components:
+  - type: Sprite
+    state: sandwich
+
+- type: entity
+  name: strawberry ice cream sandwich
+  parent: FoodFrozenBase
+  id: FoodFrozenSandwichStrawberry
+  description: Portable ice-cream in its own packaging of the strawberry variety.
+  components:
+  - type: Sprite
+    state: sandwich-strawberry
+
+- type: entity
+  name: space freezy
+  parent: FoodFrozenBase
+  id: FoodFrozenFreezy
+  description: The best icecream in space.
+  components:
+  - type: Sprite
+    state: spacefreezy
+
+- type: entity
+  name: ice cream sundae
+  parent: FoodFrozenBase
+  id: FoodFrozenSundae
+  description: A classic dessert.
+  components:
+  - type: Sprite
+    state: spacefreezy
+
+- type: entity
+  name: cornuto
+  parent: FoodFrozenBase
+  id: FoodFrozenCornuto
+  description: A neapolitan vanilla and chocolate icecream cone. It menaces with a sprinkling of caramelized nuts.
+  components:
+  - type: Sprite
+    state: cornuto
+
+# Popsicle
+
+- type: entity
+  name: orange creamsicle
+  parent: FoodFrozenBase
+  id: FoodFrozenPopsicleOrange
+  description: A classic orange creamsicle. A sunny frozen treat.
+  components:
+  - type: Sprite
+    layers:
+    - state: stick
+    - state: popsicle-base
+    - state: popsicle
+      color: orange
+  - type: Food
+    trash: FoodFrozenPopsicleTrash
+
+- type: entity
+  name: berry creamsicle
+  parent: FoodFrozenBase
+  id: FoodFrozenPopsicleBerry
+  description: A vibrant berry creamsicle. A berry good frozen treat.
+  components:
+  - type: Sprite
+    layers:
+    - state: stick
+    - state: popsicle-base
+    - state: popsicle
+      color: red
+  - type: Food
+    trash: FoodFrozenPopsicleTrash
+
+- type: entity
+  name: jumbo icecream
+  parent: FoodFrozenBase
+  id: FoodFrozenPopsicleJumbo
+  description: A luxurious icecream covered in rich chocolate. It's smaller than you remember.
```

Comment:
Same as above ("ice-cream" vs "icecream")

Reply:
What about just "ice cream"? Think thats the correct spelling...

Reply:
> 
> 
> What about just "ice cream"? Think thats the correct spelling...

I'm using ice-cream.